fastcrud/crud/fast_crud.py

```diff
@@ -420,6 +422,51 @@ async def get(
             )
         return schema_to_select(**out)
 
+    def _get_pk_dict(self, instance):
+        return {pk.name: getattr(instance, pk.name) for pk in self._primary_keys}
+
+    async def upsert(
+        self,
+        db: AsyncSession,
+        instance: Union[UpdateSchemaType, CreateSchemaType],
+        schema_to_select: Optional[type[BaseModel]] = None,
+        return_as_model: bool = False,
+    ) -> Union[BaseModel, Dict[str, Any], None]:
+        """Update the instance or create it if it doesn't exists.
+
+        Args:
+            db (AsyncSession): The database session to use for the operation.
+            instance (Union[UpdateSchemaType, type[BaseModel]]): A Pydantic schema representing the instance.
+            schema_to_select (Optional[type[BaseModel]], optional): Optional Pydantic schema for selecting specific columns. Defaults to None.
+            return_as_model (bool, optional): If True, converts the fetched data to Pydantic models based on schema_to_select. Defaults to False.
+
+        Returns:
+            BaseModel: the created or updated instance
+        """
+        _pks = self._get_pk_dict(instance)
+        schema_to_select = schema_to_select or type(instance)
+        db_instance = await self.get(
+            db,
+            schema_to_select=schema_to_select,
+            return_as_model=return_as_model,
+            **_pks,
+        )
+        if db_instance is None:
+            db_instance = await self.create(db, instance)  # type: ignore
+            db_instance = schema_to_select.model_validate(
+                db_instance, from_attributes=True
+            )
+        else:
+            await self.update(db, instance)  # type: ignore
+            db_instance = await self.get(
```

Comment:
Wondering here if we should also run a `get` and return it, we currently don't do it with update. It adds another query to the db which the user may or may not want.